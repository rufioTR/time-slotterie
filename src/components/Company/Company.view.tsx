import React from "react";

import { css, cx } from "@emotion/css";

import { TimeSlotSelectionsType } from "@root/pages/TimeSlotOverview/TimeSlotOverview.types";

import { colors } from "@root/config/colors";

import SelectedTimeRangeContainer from "../SelectedTimeRange/SelectedTimeRange.container";

export interface ICompanyProps {
  className?: string;
  title: string;
  id: number;
  selectedTimeRanges: TimeSlotSelectionsType[];
}

const Company: React.FC<ICompanyProps> = ({
  className,
  title,
  children,
  id,
  selectedTimeRanges,
}) => {
  const styles = css`
    flex: 0 0 300px;
    width: 300px;
    height: 100%;
    background-color: ${colors.white};
    color: ${colors.black};
    margin-right: 32px;
    border-radius: 4px;
    text-align: center;
    position: relative;
    display: flex;
    flex-direction: column;

    .top_content {
      height: 112px;
      padding: 32px;

      h2 {
        margin-bottom: 16px;
      }

      border-bottom: 1px solid ${colors.lightGrey};
    }

    .time_range_content {
      padding: 32px;
      height: calc(100% - 100px);
      overflow: auto;
    }
  `;

  return (
    <div className={cx(styles, className)}>
      <div className="top_content">
        <h2>{title}</h2>
        <SelectedTimeRangeContainer
          selectedTimeRanges={selectedTimeRanges}
          companyId={id}
        />
      </div>
      <div className="time_range_content">{children}</div>
    </div>
  );
};

export default Company;
