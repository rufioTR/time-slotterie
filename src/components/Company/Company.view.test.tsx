import Company from "./Company.view";
import React from "react";
import { render } from "@testing-library/react";

describe("Company", () => {
  const mockSelectedTimeRanges: React.ComponentProps<
    typeof Company
  >["selectedTimeRanges"] = [
    {
      companyId: 1,
      endTime: "2018-07-09T13:00:00.000+02:00",
      startTime: "2018-07-09T11:30:00.000+02:00",
    },
  ];

  it("should render component correctly", () => {
    // arrange
    const { asFragment } = render(
      <Company
        title="Mock Title"
        id={1}
        selectedTimeRanges={mockSelectedTimeRanges}
      >
        Mock Content
      </Company>
    );
    // act
    // assert
    expect(asFragment()).toMatchSnapshot();
  });

  it("should render component correctly without selected time range", () => {
    // arrange
    const { asFragment } = render(
      <Company
        title="Mock Title"
        id={2}
        selectedTimeRanges={mockSelectedTimeRanges}
      >
        Mock Content
      </Company>
    );
    // act
    // assert
    expect(asFragment()).toMatchSnapshot();
  });
});
