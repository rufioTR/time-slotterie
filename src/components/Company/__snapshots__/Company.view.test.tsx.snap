// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Company should render component correctly 1`] = `
<DocumentFragment>
  <div
    class="css-tkduzr"
  >
    <div
      class="top_content"
    >
      <h2>
        Mock Title
      </h2>
      <div
        class="css-yhk98z"
      >
        <div
          class="selected_range"
        >
          <div
            class="day"
          >
            Monday
          </div>
          <div>
            11:30 - 13:00
          </div>
        </div>
      </div>
    </div>
    <div
      class="time_range_content"
    >
      Mock Content
    </div>
  </div>
</DocumentFragment>
`;

exports[`Company should render component correctly without selected time range 1`] = `
<DocumentFragment>
  <div
    class="css-tkduzr"
  >
    <div
      class="top_content"
    >
      <h2>
        Mock Title
      </h2>
      <div
        class="css-yhk98z"
      >
        <div>
          nothing selected yet
        </div>
      </div>
    </div>
    <div
      class="time_range_content"
    >
      Mock Content
    </div>
  </div>
</DocumentFragment>
`;
